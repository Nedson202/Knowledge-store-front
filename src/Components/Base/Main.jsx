import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import UnsplashMention from './UnsplashMention';

class Main extends Component {
  navigateToBooks = () => window.location.replace('/books');

  renderButtons() {
    return (
      <Fragment>
        <button
          type="button"
          className="btn btn-outline-primary"
          data-toggle="modal"
          data-target="#LoginFormModal"
        >
          <ion-icon name="log-in" />
          Login to get started
          {' '}
        </button>
        <button
          type="button"
          className="btn btn-outline-primary"
          data-toggle="modal"
          data-target="#SignUpFormModal"
        >
          <ion-icon name="log-in" />
          Signup
          {' '}
        </button>
      </Fragment>
    );
  }

  renderExploreButton() {
    return (
      <Fragment>
        <button
          type="button"
          className="btn btn-outline-primary"
          onClick={this.navigateToBooks}
        >
          <ion-icon name="paper-plane" />
          {' '}
          Explore
        </button>
      </Fragment>
    );
  }

  renderMainContent() {
    return (
      <Fragment>
        <h2 className="product-title">The Knowledge Store</h2>
        <blockquote>
          <p className="quotation">
            Wisdom is not a product of schooling
            but of the lifelong attempt to acquire it.
          </p>
          <footer>â€• Albert Einstein</footer>
        </blockquote>
      </Fragment>
    );
  }

  render() {
    const { isAuthenticated } = this.props;
    return (
      <div className="auth-buttons">
        <div>
          {this.renderMainContent()}
          <div className="auth-action-buttons">
            {this.renderExploreButton()}
            {!isAuthenticated && this.renderButtons()}
          </div>
        </div>

        <UnsplashMention />
      </div>
    );
  }
}

Main.propTypes = {
  isAuthenticated: PropTypes.bool,
};

Main.defaultProps = {
  isAuthenticated: false,
};

export default Main;
