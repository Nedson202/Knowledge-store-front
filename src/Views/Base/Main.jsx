import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import UnsplashMention from './UnsplashMention';
import { BOOKS_PATH } from '../../settings';

class Main extends Component {
  navigateToBooks = () => window.location.replace(BOOKS_PATH);

  renderButtons() {
    const { isAuthenticated } = this.props;

    if (isAuthenticated) {
      return;
    }

    return (
      <Fragment>
        <button
          className="btn btn-outline-primary"
          data-toggle="modal"
          data-target="#LoginFormModal"
          data-testid="index-login-action"
          type="button"
        >
          <ion-icon name="log-in" />
          Login to get started
        </button>
        <button
          className="btn btn-outline-primary"
          data-toggle="modal"
          data-target="#SignUpFormModal"
          data-testid="index-signup-action"
          type="button"
        >
          <ion-icon name="log-in" />
          Signup
        </button>
      </Fragment>
    );
  }

  renderExploreButton() {
    return (
      <button
        className="btn btn-outline-primary"
        onClick={this.navigateToBooks}
        type="button"
      >
        <ion-icon name="paper-plane" />
          Explore
      </button>
    );
  }

  renderMainContent() {
    return (
      <Fragment>
        <h2 className="app-title">The Knowledge Store</h2>
        <blockquote>
          <p className="app-index-quote">
            Wisdom is not a product of schooling
            but of the lifelong attempt to acquire it.
          </p>
          <footer className="app-index-quote-author">â€• Albert Einstein</footer>
        </blockquote>
      </Fragment>
    );
  }

  render() {
    return (
      <div className="app-container">
        <div>
          {this.renderMainContent()}
          <div className="auth-action-buttons">
            {this.renderExploreButton()}
            {this.renderButtons()}
          </div>
        </div>

        <UnsplashMention />
      </div>
    );
  }
}

Main.propTypes = {
  isAuthenticated: PropTypes.bool,
};

Main.defaultProps = {
  isAuthenticated: false,
};

export default Main;
