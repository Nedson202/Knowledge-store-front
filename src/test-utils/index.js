import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

// eslint-disable-next-line react/prop-types
export const AllProviders = ({ children, customStore }) => (
  <MockedProvider removeTypename>
    <Provider store={customStore || store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  </MockedProvider>
);

export * from '@testing-library/react';
