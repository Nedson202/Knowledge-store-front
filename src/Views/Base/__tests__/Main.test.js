import React from 'react';

import { AllProviders, render } from 'test-utils';

import Main from '..';

describe('Main', () => {
  it('should render action buttons', () => {
    const {
      container, asFragment, getByText,
    } = render(
      <AllProviders>
        <Main />
      </AllProviders>
    );

    getByText('Login to get started');
    getByText('Signup');
    getByText('Explore');

    expect(asFragment()).toMatchSnapshot();
    expect(container).toBeVisible();
  });

  it('should render page header summary', () => {
    const { getByText } = render(
      <AllProviders>
        <Main />
      </AllProviders>
    );

    getByText(
      'Wisdom is not a product of schooling but of the lifelong attempt to acquire it.'
    );
    getByText('â€• Albert Einstein');
    getByText('Explore');
  });

  it('should not render auth buttons if signed in', () => {
    const props = {
      isAuthenticated: true,
    };
    const { queryByTestId } = render(
      <AllProviders>
        <Main {...props} />
      </AllProviders>
    );

    const loginButton = queryByTestId('index-login-button');
    const signupButton = queryByTestId('index-signup-button');

    expect(loginButton).toBeNull();
    expect(signupButton).toBeNull();
  });
});
