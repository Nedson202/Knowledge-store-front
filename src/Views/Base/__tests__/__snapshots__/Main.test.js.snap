// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render action buttons 1`] = `
<DocumentFragment>
  <div>
    <div
      id="background-layout"
    />
    <div
      class="app-container"
    >
      <div>
        <h2
          class="app-title"
        >
          The Knowledge Store
        </h2>
        <blockquote>
          <p
            class="quotation"
          >
            Wisdom is not a product of schooling but of the lifelong attempt to acquire it.
          </p>
          <footer>
            â€• Albert Einstein
          </footer>
        </blockquote>
        <div
          class="auth-action-buttons"
        >
          <button
            class="btn btn-outline-primary"
            type="button"
          >
            <ion-icon
              name="paper-plane"
            />
            Explore
          </button>
          <button
            class="btn btn-outline-primary"
            data-target="#LoginFormModal"
            data-testid="index-login-action"
            data-toggle="modal"
            type="button"
          >
            <ion-icon
              name="log-in"
            />
            Login to get started
          </button>
          <button
            class="btn btn-outline-primary"
            data-target="#SignUpFormModal"
            data-testid="index-signup-action"
            data-toggle="modal"
            type="button"
          >
            <ion-icon
              name="log-in"
            />
            Signup
          </button>
        </div>
      </div>
      <a
        class="unsplash-mention"
        href="https://unsplash.com/@martinadams?utm_medium=referral&utm_campaign=photographer-credit&utm_content=creditBadge"
        rel="noopener noreferrer"
        target="_blank"
        title="Download free do whatever you want high-resolution photos from Martin Adams"
      >
        <span
          style="display: inline-block; padding: 2px 3px;"
        >
          <svg
            style="height: 12px; width: auto; position: relative; vertical-align: middle; top: -2px; fill: black;"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
               unsplash - logo
            </title>
            <path
              d="M10 9V0h12v9H10zm12 5h10v18H0V14h10v9h12v-9z"
            />
          </svg>
        </span>
        <span
          style="display: inline-block; padding: 2px 3px;"
        >
          Martin Adams
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
